// react-skroll/lib/contextProviderShape.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

exports.default = _react.PropTypes.shape({
  position: _react.PropTypes.number,
  positionRatio: _react.PropTypes.number,
  start: _react.PropTypes.number,
  end: _react.PropTypes.number,
  viewHeight: _react.PropTypes.number,
  scrollHeight: _react.PropTypes.number,
  ready: _react.PropTypes.bool,
  onStart: _react.PropTypes.bool,
  onMiddle: _react.PropTypes.bool,
  onEnd: _react.PropTypes.bool,
  children: _react.PropTypes.arrayOf(_react.PropTypes.object),

  autoFrame: _react.PropTypes.bool,
  autoScroll: _react.PropTypes.bool,

  originalPosition: _react.PropTypes.number,
  changedPosition: _react.PropTypes.number,
  timeStamp: _react.PropTypes.number,
  scrolling: _react.PropTypes.bool,
  wheeling: _react.PropTypes.bool,
  touching: _react.PropTypes.bool,
  moving: _react.PropTypes.bool,
  resting: _react.PropTypes.bool,
  // touches: PropTypes.object,

  node: _react.PropTypes.object,

  setNode: _react.PropTypes.func,
  unsetNode: _react.PropTypes.func,
  handleScroll: _react.PropTypes.func,
  handleWheel: _react.PropTypes.func,
  handleTouchStart: _react.PropTypes.func,
  handleTouchMove: _react.PropTypes.func,
  handleTouchEnd: _react.PropTypes.func,
  scrollTo: _react.PropTypes.func,
  scrollToPosition: _react.PropTypes.func,
  scrollToByIndex: _react.PropTypes.func,
  scrollToTop: _react.PropTypes.func,
  scrollToBottom: _react.PropTypes.func,
  scrollToName: _react.PropTypes.func,
  scrollToElement: _react.PropTypes.func,
  setStateMove: _react.PropTypes.func,
  setStateRest: _react.PropTypes.func,
  setStateStart: _react.PropTypes.func,
  setStateEnd: _react.PropTypes.func
});
